packages/flutter/lib/src/gestures/events.dart

```diff
@@ -323,11 +362,50 @@ abstract class PointerEvent extends Diagnosticable {
   /// the difference between the 2 events in that case.
   final bool synthesized;
 
+  /// The transformation used to transform this event from the global coordinate
+  /// space into the coordinate space of the event receiver.
+  ///
+  /// This value affects what is returned by [localPosition] and [localDelta].
+  /// If this value is null, it is treated as the identity transformation.
+  ///
+  /// Unlike a paint transform, this transform usually does not contain any
+  /// "perspective" components, meaning that the third row and the third column
+  /// of the matrix should be equal to "0, 0, 1, 0". This ensures that
+  /// [localPosition] describes the point in the local coordinate system of the
+  /// event receiver at which the user is actually touching the screen.
+  ///
+  /// See also:
+  ///
+  ///  * [transformed], which transforms this event into a different coordinate
+  ///    space.
+  final Matrix4 transform;
+
+  /// The original [PointerEvent] before it was transformed by [transform].
+  ///
+  /// If [transform] is null or the identity transformation this may be null.
+  final PointerEvent original;
```

Comment:
What is this used for? (might want to document).  Also, if the event is transformed multiple times, should it be the version of the event before the first transform, or before the most recent transform?

Reply:
Clarified and added.